homeassistant/components/synology_dsm/__init__.py

```diff
@@ -170,7 +178,8 @@ def _async_migrator(entity_entry: entity_registry.RegistryEntry):
     api = SynoApi(hass, entry)
     try:
         await api.async_setup()
-    except SynologyDSMRequestException as err:
+    except (SynologyDSMLoginFailedException, SynologyDSMRequestException) as err:
```

Comment:
Why adding `SynologyDSMLoginFailedException` here ?

I think `SynologyDSMRequestException` should be enough.

Reply:
because during boot (_after reboot or scheduled downtime_) DSM it self is already reachable (_no connection error_), but still login is unavailable (_login failed_) because DSM says "still booting, please try later"